src/components/carousel/__tests__/index.spec.js

```diff
@@ -0,0 +1,89 @@
+import {map} from 'lodash';
+import React from 'react';
+import {render, fireEvent} from '@testing-library/react-native';
+import {Text, View} from 'react-native';
+import {Constants} from '../../../helpers';
+import Carousel from '../index';
+
+const numberOfPagesShown = 5;
+const getEventData = ({y = 0, x = 0}) => {
+  return {
+    nativeEvent: {
+      contentOffset: {y, x}
+    }
+  };
+};
+const onChangePageMock = jest.fn();
+const onScrollMock = jest.fn();
+
+const TestCase = props => {
+  return (
+    <Carousel 
+      testID={'carousel'}
+      onChangePage={onChangePageMock}
+      onScroll={onScrollMock}
+      {...props}
+    >
+      {map([...Array(numberOfPagesShown)], (_, index) => (
+        <Page key={index}>
+          <Text testID={`page-${index}`}>Page #{index}</Text>
+        </Page>
+      ))}
+    </Carousel>
+  );
+};
+
+const Page = ({children, ...others}) => {
+  return (
+    <View {...others} style={{flex: 1}}>
+      {children}
+    </View>
+  );
+};
+
+describe('Carousel render tests', () => {
+  
+  describe('initialPage', () => {
+    it('should be set to default initialPage', () => {
+      const component = render(<TestCase/>);
+      const scrollView = component.getByTestId('carousel.scrollView');
+      
+      expect(scrollView.props.contentOffset.x).toBe(0);
+    });
+
+    it('should be set to initialPage = 2', () => {
+      const component = render(<TestCase initialPage={2}/>);
+      const scrollView = component.getByTestId('carousel.scrollView');
+      
+      expect(scrollView.props.contentOffset.x).toBe(Constants.screenWidth * 2);
+    });
+  });
+
+  describe('onScroll', () => {
+    it('should trigger onScroll from the second scroll', () => {
+      const component = render(<TestCase/>);
+      const scrollView = component.getByTestId('carousel.scrollView');
+
+      fireEvent.scroll(scrollView, getEventData({x: Constants.screenWidth})); //NOTE: first scroll doesn't fire onScroll
```

Comment:
Please add it to the new scrollView utils and use it accordingly

Reply:
Done